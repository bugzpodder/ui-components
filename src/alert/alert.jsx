// @flow
import CheckCircleIcon from "@material-ui/icons/CheckCircle";
import ErrorIcon from "@material-ui/icons/Error";
import InfoIcon from "@material-ui/icons/Info";
import React, { type Node } from "react";
import SnackbarContent from "@material-ui/core/SnackbarContent";
import WarningIcon from "@material-ui/icons/Warning";
import classNames from "classnames";
import styles from "./alert.module.scss";

type Props = {
  /** The message to display in the alert */
  message: Node,
  /** Classname applied to the alert component */
  className?: string,
  /**
   * Provides classNames to the alert and its sub-components. Options include:
   *  - `root`
   *
   *  - `content`
   */
  classes?: AlertClasses,
  /** The color styling of the alert. Defaults to `error` */
  color?: AlertColor,
  /** When "center", aligns the Alert's messages horizontally */
  textAlign?: string,
  /** When "text", changes the styling of the alert */
  variant?: string,
};

const colorIcon = {
  success: CheckCircleIcon,
  warning: WarningIcon,
  error: ErrorIcon,
  info: InfoIcon,
};

/**
 * `Alert is a styled alert pane for UI components`
 */
export const Alert = (props: Props) => {
  const {
    textAlign = "", className = "", classes = {}, color = "error", message, variant = "", ...alertProps
  } = props;
  const Icon = colorIcon[color];
  return (
    <div
      data-testid="alert"
      className={classNames(styles.alertContainer, classes.root)}
    >
      <SnackbarContent
        className={classNames(styles[color], styles[variant], `alert-${color}`, className, classes.content)}
        classes={{
          message: styles[textAlign],
        }}
        message={(
          <span>
            <Icon className={styles.alertIcon} />
            {message}
          </span>
)}
        {...alertProps}
      />
    </div>
  );
};
