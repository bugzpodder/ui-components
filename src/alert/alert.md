## Examples

```js
const styles = require("@grail/components/src/utils/example-styles.module.scss");
<div className={styles.container}>
	<h2> Default </h2>
	<Alert className={styles.spacing} color="success" message="This is successful!" />
	<Alert className={styles.spacing} color="info" message="This is some info!" />
	<Alert className={styles.spacing} color="warning" message="This is a warning!" />
	<Alert className={styles.spacing} message="This is an error!" />
	<br />
	<h2> Flat </h2>
	<Alert className={styles.spacing} variant="flat" color="success" message="This is successful!" />
	<Alert className={styles.spacing} variant="flat" color="info" message="This is some info!" />
	<Alert className={styles.spacing} variant="flat" color="warning" message="This is a warning!" />
	<Alert className={styles.spacing} variant="flat" message="This is an error!" />
</div>;
```
