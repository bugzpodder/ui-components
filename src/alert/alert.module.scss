@import "~@grail/lib/constants/colors.scss";


@mixin flatten-colors($background-color, $background-percentage, $text-color, $text-percentage) {
	background-color: lighten($background-color, $background-percentage);
	color: darken($text-color, $text-percentage);
}

.alertContainer {

	.flat {
		box-shadow: none;
	}

	.error {
		background-color: $grail-red-color;
		color: #FFF;
		&.flat {
			@include flatten-colors($grail-red-color, 42%, $grail-red-color, 20%);
		}
	}

	.info {
		background-color: $grail-blue-color;
		color: #FFF;
		&.flat {
			@include flatten-colors($grail-blue-color, 58%, $grail-blue-color, 10%);
		}
	}

	.success {
		background-color: $grail-green-color;
		color: #FFF;
		&.flat {
			@include flatten-colors($grail-green-color, 62%, $grail-green-color, 15%);
		}
	}

	.warning {
		background-color: $grail-gold-color;
		color: #FFF;
		&.flat {
			@include flatten-colors($grail-gold-color, 48%, $grail-contrast-color, 15%);
		}

	}

	.center {
		margin: 0 8px;
		width: 100%;
		text-align: center;
	}

	.alertIcon {
		margin-right: 5px;
		vertical-align: middle;
	}
}
