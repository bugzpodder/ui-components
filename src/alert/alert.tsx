import CheckCircleIcon from "@material-ui/icons/CheckCircle";
import ErrorIcon from "@material-ui/icons/Error";
import InfoIcon from "@material-ui/icons/Info";
import React, { ReactNode } from "react";
import SnackbarContent from "@material-ui/core/SnackbarContent";
import WarningIcon from "@material-ui/icons/Warning";
import classNames from "classnames";
import styles from "./alert.module.scss";

type AlertClasses = {
  root?: string;
  content?: string;
};

type Props = {
  /** The message to display in the alert */
  message: ReactNode;
  /** Classname applied to the alert component */
  className?: string;
  /**
   * Provides classNames to the alert and its sub-components. Options include:
   *  - `root`
   *
   *  - `content`
   */
  classes?: AlertClasses;
  /** The color styling of the alert. Defaults to `error` */
  color?: string;
  /** When "center", aligns the Alert's messages horizontally */
  textAlign?: string;
  /** When "text", changes the styling of the alert */
  variant?: string;
};

const colorIcon = {
  success: CheckCircleIcon,
  warning: WarningIcon,
  error: ErrorIcon,
  info: InfoIcon,
};

/**
 * `Alert is a styled alert pane for UI components`
 */
export const Alert: React.FC<Props> = props => {
  const {
    textAlign = "",
    className = "",
    classes = {},
    color = "error",
    message,
    variant = "",
    ...alertProps
  } = props;
  const Icon = colorIcon[color];
  return (
    <div
      data-testid="alert"
      className={classNames(styles.alertContainer, classes.root)}
    >
      <SnackbarContent
        data-testid="content"
        className={classNames(
          styles[color],
          styles[variant],
          `alert-${color}`,
          className,
          classes.content,
        )}
        classes={{
          message: styles[textAlign],
        }}
        message={
          <span data-testid={`${color}-alert-message`}>
            <Icon
              data-testid={`${color}-alert-icon`}
              className={styles.alertIcon}
            />
            {message}
          </span>
        }
        {...alertProps}
      />
    </div>
  );
};
