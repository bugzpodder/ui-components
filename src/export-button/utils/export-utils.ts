import {
  ExportableColumn,
  ReportOptions,
  toDelimitedReport,
} from "@grailbio/lib";
import { saveAs } from "file-saver";

export const generateReport = (
  fileName: string,
  columns: ExportableColumn<any>[],
  data: Record<string, any>[],
  options: ReportOptions = {},
): void => {
  const output = toDelimitedReport(columns, data, options);
  const { fileMimeType = "text/csv" } = options;
  const blob = new Blob([output], { type: `${fileMimeType};charset=utf-8;` });
  try {
    saveAs(blob, fileName, true);
  } catch (error) {
    console.error("error saving blob", error);
  }
};
