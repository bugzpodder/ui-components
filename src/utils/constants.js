// @flow
export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  value: suggestion.label,
  label: suggestion.label,
}));
