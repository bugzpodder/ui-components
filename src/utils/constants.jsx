// @flow
import React from "react";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const SERGEY =
  "https://media.licdn.com/dms/image/C5603AQGlBIPst1V4Ew/profile-displayphoto-shrink_800_800/0?e=1535587200&v=beta&t=2V2knyEhbTrhRWzgydsXLZqiX8gnoD1poj8jGeFpA0w";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];
