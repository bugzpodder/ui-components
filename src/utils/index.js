// @flow
export { COUNTRIES, SERGEY } from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
