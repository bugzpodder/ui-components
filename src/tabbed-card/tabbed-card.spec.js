// @flow
import "jest-dom/extend-expect";
import React from "react";
import { Alert } from "../alert/index";
import { TabbedCard } from "./index";
import { TestWrapper } from "../utils";
import { cleanup, render } from "react-testing-library";

afterEach(cleanup);

const validHeaderTabs = [
  {
    label: "Tab One",
    value: "one",
    id: "tab-one",
    content: <Alert
      color="success"
      message="Tab One!"
    />,
  },
  {
    label: "Tab Two",
    value: "two",
    id: "tab-two",
    content: <Alert
      color="success"
      message="Tab Two!"
    />,
  },
];

const headerActions = [
  {
    content: "Test Action",
  },
];

test("render tabbed card", () => {
  const { container, getByTestId } = render(
    <TestWrapper>
      <TabbedCard
        title="Test Card"
        value="one"
        headerTabs={validHeaderTabs}
        headerActions={headerActions}
        onChange={() => {}}
      />
    </TestWrapper>,
  );
  expect(getByTestId("alert")).toHaveTextContent("Tab One!");
  expect(getByTestId("tab-one")).toHaveAttribute("aria-selected", "true");
  expect(getByTestId("alert")).not.toHaveTextContent("Tab Two!");
  expect(getByTestId("tab-two")).toHaveAttribute("aria-selected", "false");
  expect(container).toMatchSnapshot();
});
