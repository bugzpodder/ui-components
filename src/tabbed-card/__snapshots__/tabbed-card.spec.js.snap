// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty tabbed card 1`] = `
<div>
  <div
    class="tabbedCardContainer"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiCard-root stickyHeader pageHeader"
    >
      <div
        class="MuiCardHeader-root cardHeader"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-headline MuiCardHeader-title header-title"
          />
          <span
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiCardHeader-subheader"
          />
        </div>
        <div
          class="MuiCardHeader-action headerActions"
        >
          <div
            class="headerActions"
          />
        </div>
      </div>
      <div
        class="MuiTabs-root"
        data-testid="card-tabs"
      >
        <div
          class="MuiTabs-flexContainer"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            role="tablist"
            style="margin-bottom: 0px;"
          >
            <div
              class="MuiTabs-flexContainer"
            />
            <span
              class="TabIndicator-root-202 TabIndicator-colorPrimary-203 MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`render tabbed card 1`] = `
<div>
  <div
    class="tabbedCardContainer"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiCard-root stickyHeader pageHeader"
    >
      <div
        class="MuiCardHeader-root cardHeader"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-headline MuiCardHeader-title header-title"
          >
            Test Card
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiCardHeader-subheader"
          />
        </div>
        <div
          class="MuiCardHeader-action headerActions"
        >
          <div
            class="headerActions"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-flat"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Test Action
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiTabs-root"
        data-testid="card-tabs"
      >
        <div
          class="MuiTabs-flexContainer"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed"
            role="tablist"
            style="margin-bottom: 0px;"
          >
            <div
              class="MuiTabs-flexContainer"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root headerTab MuiTab-textColorPrimary MuiTab-selected"
                data-testid="tab-one"
                id="tab-one"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  <span
                    class="MuiTab-labelContainer"
                  >
                    <span
                      class="MuiTab-label"
                    >
                      Tab One
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root headerTab MuiTab-textColorPrimary"
                data-testid="tab-two"
                id="tab-two"
                role="tab"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiTab-wrapper"
                >
                  <span
                    class="MuiTab-labelContainer"
                  >
                    <span
                      class="MuiTab-label"
                    >
                      Tab Two
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="TabIndicator-root-118 TabIndicator-colorPrimary-119 MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="alertContainer"
        data-testid="alert"
      >
        <div
          class="MuiTypography-root MuiTypography-body1 MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root success alert-success"
          role="alertdialog"
        >
          <div
            class="MuiSnackbarContent-message"
          >
            <span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root alertIcon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              Tab One!
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
